File: fe/lib/models/ca-truc.dart
// ignore_for_file: file_names

import 'bai-xe.dart';
import 'user.dart';

class CaTruc {
  int? id;
  int? idNv;
  User? nhanVien;
  int? idBx;
  BaiXe? baiXe;
  String? noiDung;
  String? startTime;
  String? endTime;
  String? startClick;
  String? endClick;
  int? soLgVe;
  int? ca;
  int? tienVe;
  int? status;
  CaTruc({
    this.id,
    this.idNv,
    this.nhanVien,
    this.idBx,
    this.baiXe,
    this.noiDung,
    this.startTime,
    this.endTime,
    this.startClick,
    this.endClick,
    this.soLgVe,
    this.ca,
    this.tienVe,
    this.status,
  });
  CaTruc.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        idNv = json['idNv'],
        nhanVien = User.fromJson(json['nhanVien']),
        idBx = json['idBx'],
        baiXe = BaiXe.fromJson(json['baiXe']),
        noiDung = json['noiDung'],
        startTime = json['startTime'],
        endTime = json['endTime'],
        startClick = json['startClick'],
        endClick = json['endClick'],
        soLgVe = json['soLgVe'],
        tienVe = json['tienVe'],
        status = json['status'];
  Map<String, dynamic> toJson() => {
        'idNv': idNv,
        'idBx': idBx,
        'noiDung': noiDung,
        'ca': ca,
        'startTime': startTime,
        'endTime': endTime,
        'startClick': startClick,
        'endClick': endClick,
        'soLgVe': soLgVe,
        'tienVe': tienVe,
        'status': status,
      };
}

File: BE/src/main/java/com/haivn/common_api/CaTruc.java
package com.haivn.common_api;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "ca_truc")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class CaTruc  extends BaseEntity{
    @Column(name = "idNv")
    private Long idNv;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="idNv",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private NguoiDung nhanVien;
    @Column(name = "idBx")
    private Long idBx;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="idBx",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private BaiXe baiXe;
    @Column(name = "noi_dung")
    private String noiDung;
    @Column(name = "ca")
    private Short ca;
    @Column(name = "start_time")
    private Timestamp startTime;
    @Column(name = "end_time")
    private Timestamp endTime;
    @Column(name = "start_click")
    private Timestamp startClick;
    @Column(name = "end_click")
    private Timestamp endClick;
    @Column(name = "so_lg_ve")
    private Long soLgVe;
    @Column(name = "tien_ve")
    private Long tienVe;
    @Column(name = "status")
    private Short status;
}
